### PR TITLE
TST refactor tests

### PR DESCRIPTION
**Pull request by eickenberg**
Consolidate tests

They currently run in 7:51 on cpu on my machine

TODO:
- [x] add cuda tests back
- [ ] (potentially undo the cartesian product of isign and fftshift and distribute these settings randomly throughout the other configs) _won't do this_

### CODE REVIEW

File: tests/test_t1_forward.py [hunk 1]
```diff
@@ -39,77 +39,35 @@ def check_t1_forward(N: int, dim: int, device: str) -> None:
...
+    (10, 3),
+    (37, 3),
 ]
 
 
-@pytest.mark.parametrize("N", Ns)
-def test_t1_3D_forward_CPU(N: int) -> None:
-    check_t1_forward(N, 3, "cpu")
+@pytest.mark.parametrize("N_and_dim", Ns_and_dims)
+def test_t1_forward_CPU(N_and_dim) -> None:
```
Comment (WardBrian) [inline tests/test_t1_forward.py:65]: Can you split these so it is two arguments `N` and `dim` for both of these tests?

**Review by WardBrian — COMMENTED** [review]
Thanks for doing this!

File: tests/test_t1_forward.py (hunk 1 above)
Comment (eickenberg) [reply tests/test_t1_forward.py:65]: yes, and then `@pytest.mark.parametrize("N, dims", Ns_and_dims)` ?

**Review by WardBrian — APPROVED** [review]
This is a dramatic improvment!

The CUDA tests seem quite slow (~3x the CPU times), maybe worth looking into another time?

**Pull request merged**